modules/addons/external-dns/variables.tf

```diff
@@ -0,0 +1,32 @@
+variable "cluster_name" {
+}
+
+variable "k8s_host" {
+}
+
+variable "k8s_cluster_ca_certificate" {
+}
+
+variable "k8s_client_token" {
+}
+
+variable "tsb_fqdn" {
+}
+
+variable "dns_provider" {
+}
+
+variable "dns_zone" {
+}
+
+# Provider specific variables. All of them should be optional to allow
+# configuring only the ones for the provider being used.
+
+variable "google_project" {
+  default = ""
+}
+
+variable "google_service_account_key" {
```

Comment:
depends how you want to twist, but we should account the implementation for the other two clouds as well, i.e. we cannot have a generic function that only can be used by GCP. However technically external-dns is a very unique function, so perhaps we have a cloud specific external-dns module that also takes care about managing the dependencies for the module itself

Reply:
This is a generic module that should be expanded with the other cloud provider parameters.
I already replied to the module structure thing in previous comments.

Regarding supporting all clouds, yes, we want to support both, but it's not needed for this PR. [You already said this is an OK approach and that you'd do the rest](https://github.com/tetrateio/tetrate-service-bridge-sandbox/issues/184#issuecomment-1364102871) so I count on that after this PR is merged.

Reply:
i think it is important to keep the intention of this repo in line to the point behind it is there to support multi-cloud demos, when integration is being done I also do think how can I fit AWS or Azure in the same time, not later as technically we always do multi-cloud demos so I cannot ignore the matter, so I think it will be fair for the important function like this ... consider how it can be done to account at least Azure or AWS, so the PR does need to include AWS or Azure, but it should aligned for multi-cloud usage pattern.

Reply:
This is not core funcitonality; it is an add-on and it's not mandatory for demos. You have survived demos till now without this, so, why shouldn't we be able to implement it incrementally?

Also, the code in this PR is fully prepared to be replicated in other clouds without major changes. it doesn't follow the structure you had envisioned, but there are good reasons and arguments to have it this way. The only thing that could change is moving the `credentials` module inside the `modules` folder, but it still needs to be a separate module with a different execution, because, as I said before, it has to be executed only once, but external-dns needs to be installed for each cluster.

Reply:
the execution can store the output in `outputs` and execute only once based on the existence of the file, after on file can retrieved and stored a `data` for the given `module`.